training/src/app/chuck-noris/search/search.component.ts

```diff
@@ -0,0 +1,46 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { debounceTime, distinctUntilChanged, Subject, takeUntil, tap } from "rxjs";
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+
+@Component({
+	selector: 'app-search',
+	templateUrl: './search.component.html',
+	styleUrls: ['./search.component.scss']
+})
+export class SearchComponent implements OnInit {
+	public searchForm: FormGroup;
+	@Output() public generateJokesFromSearch: EventEmitter<string> = new EventEmitter<string>();
+	@Output() public generateJokesFromClick: EventEmitter<string>  = new EventEmitter<string>();
```

Comment:
whats this event? why does search component needs to know about buttons and clicks? i think what u tried to do is event that emit the search value, it doesnt needs to know that the parent will do something with click with it